MoltenVK/MoltenVK/GPUObjects/MVKQueue.mm

```diff
@@ -705,6 +705,12 @@
 		setConfigurationResult(_presentInfo[i].presentableImage->presentCAMetalDrawable(mtlCmdBuff, _presentInfo[i]));
 	}
 
+	if (_queue->_queueFamily->getIndex() == getMVKConfig().defaultGPUCaptureScopeQueueFamilyIndex &&
+		_queue->_index == getMVKConfig().defaultGPUCaptureScopeQueueIndex) {
+		getDevice()->stopAutoGPUCapture(MVK_CONFIG_AUTO_GPU_CAPTURE_SCOPE_ON_DEMAND);
+		getDevice()->startAutoGPUCapture(MVK_CONFIG_AUTO_GPU_CAPTURE_SCOPE_ON_DEMAND, _queue->getMTLCommandQueue());
+	}
```

Comment:
What happens when you restart the capture on each frame? Metal typically refuses to overwrite the capture file.

Reply:
It doesn't restart on each frame. If there's more bytes in the pipe, `stopAutoGPUCapture` consumes a byte and just returns, and then `startAutoGPUCapture` also just bails out because it's still capturing. That way you can write N bytes to capture N consecutive frames. Once the capture is completed, you just move the capture file somewhere else and you're good to make another one.

Reply:
Ah. I see now. Very clever.